virttest/qemu_monitor.py

```diff
@@ -3635,3 +3636,46 @@ def query_sgx_capabilities(self):
         cmd = "query-sgx-capabilities"
         self.verify_supported_cmd(cmd)
         return self.cmd(cmd)
+
+    def query_sev(self):
+        """
+        Query SEV information. (Since 2.12)
+
+        :return: Information about SEV
+        """
+        cmd = "query-sev"
+        self.verify_supported_cmd(cmd)
+        return self.cmd(cmd)
+
+    def query_sev_capabilities(self):
+        """
+        Query SEV capabilities. (Since 2.12)
+
+        :return: SEV capability objects.
+        """
+        cmd = "query-sev-capabilities"
+        self.verify_supported_cmd(cmd)
+        return self.cmd(cmd)
+
+    def query_sev_launch_measure(self):
+        """
+        Query the SEV guest launch information. (Since 2.12)
+
+        :return: The SEV launch measure info for the guest.
+        """
+        cmd = "query-sev-launch-measure"
+        self.verify_supported_cmd(cmd)
+        return self.cmd(cmd)
+
+    def query_sev_attestation_report(self):
+        """
+        Query SEV attestation report. (Since 6.1)
+        Note that it needs to generate a random 16 bytes value encoded
+        in base64 to query attestation report.
+
+        :return: The SEV attestation report objects.
+        """
+        cmd = "query-sev-attestation-report"
+        mnonce = base64.b64encode(os.urandom(16)).decode()
```

Comment:
as mnonce will be included in the report, do we want to set the mnonce outside?
e.g. set it in test case, call this function, verify if the mnonce is included, or if the data from the report is returned in an expected way

Reply:
Thanks Zhenchao! Yes we can set mnonce outside, my concern is if anyone who not familiar about this function, they may not generate a mnonce with the right encoding method, so I put it in this function. By the way, could we make this an optional input, if no input, then we generate  mnonce inside it.

Reply:
If no input, currently, it means this option is not required by the qmp cmd, e.g. speed for block commit, qemu will use a default value, in that case, we can cover the scenario that using default values.

No worry. As a public interface, we need to write the doc string carefully to tell users how to call it, if they use it in a wrong way, then they should get an error or something likely, which is expected